server/src/main/java/org/elasticsearch/rest/ChunkedRestResponseBody.java

```diff
@@ -43,6 +44,21 @@ public interface ChunkedRestResponseBody {
      */
     boolean isDone();
 
+    /**
+     * @return true if this is the last chunked body in the response.
+     */
+    boolean isEndOfResponse();
+
+    /**
+     * Asynchronously retrieves the next part of the body. Note that this is called on a transport thread, so implementations must take care
+     * to dispatch any nontrivial work elsewhere.
+     *
+     * @param listener Listener to complete with the next part of the body. By the point this is called we have already started to send
+     *                 the body of the response, so there's no good ways to handle an exception here. Completing the listener exceptionally
+     *                 will log an error, abort sending the response, and close the HTTP connection.
+     */
+    void getContinuation(ActionListener<ChunkedRestResponseBody> listener);
```

Comment:
ES|QL's operations use this:
```
    /**
     * An operator can be blocked on some action (e.g. waiting for some resources to become available).
     * If so, it returns a future that completes when the operator becomes unblocked.
     * If the operator is not blocked, this method returns {@link #NOT_BLOCKED} which is an already
     * completed future.
     */
    default SubscribableListener<Void> isBlocked() {
        return NOT_BLOCKED;
    }

    SubscribableListener<Void> NOT_BLOCKED = SubscribableListener.newSucceeded(null);
```

We should probably both do the same thing because we're solving the same fundamental problem. 

Well, I guess we are different in a critical way - yours is called on the transport thread and ES|QL's are called on ESQL's thread pool.

Reply:
I think the main difference is that Netty already has all the machinery needed to handle the continuation, no need to involve a `SubscribableListener`, a plain old `ActionListener` that calls `Channel#writeAndFlush` is enough.